doc/compilation.dox

```diff
@@ -0,0 +1,383 @@
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+/**
+
+\htmlonly
+
+<style type="text/css" media="screen, projection">
+pre {
+    padding: 0.5em;
+    border: 1px dashed #2f6fab;
+    color: black;
+    background-color: #f9f9f9;
+    line-height: 1em;
+}
+
+#note {
+    position: relative;
+}
+#note span {
+    display: none;
+}
+#note a span {
+    display: none;
+    color: #FFFFFF;
+}
+#note a span span {
+    display: none;
+}
+#note a:hover span span {
+    display: none;
+}
+#note a:hover span {
+    display: block;
+    position: absolute;
+    width: 20em;
+    background-color: #aaa;
+    left: 1em;
+    top: -5px;
+    color: #FFFFFF;
+    padding: 5px; }
+</style>
+\endhtmlonly
+
+\page compilation Building RTNeuron
+
+The only officially supported operating systems is Ubuntu and its derivates, for
+which this guide is written. For any other Linux distribution the instructions
+are almost the same, the biggest difference being the installation of
+dependencies.Windows and OS/X are not officially supported.
+
+\section rtneuron_dependencies Dependencies
+
+RTNeuron dependencies are split into two groups, those that need to be
+preinstalled by the user and those that can be built by RTNeuron's compilation
+scripts. Only dependencies that require manual installation are listed, for all
+the others the instructions will rely on the automatic dependency building.
+Dependencies for experimental or unmaintained features are not considered in
+this guide.
+
+For convenience the complete apt-get command to use in Ubuntu with Python3 to
+install the required and some recommended optional dependencies is the
+following:
+
+\verbatim
+sudo apt-get install cmake g++ git bison flex libboost-dev libboost-filesystem-dev libboost-iostreams-dev \
+    libboost-regex-dev libboost-program-options-dev libboost-python-dev libboost-system-dev libboost-serialization-dev \
+    libboost-test-dev libboost-thread-dev libhdf5-serial-dev libopenscenegraph-dev mesa-common-dev qtdeclarative5-dev \
+    qml-module-qtquick-controls qml-module-qtquick-layouts qml-module-qtquick-dialogs libpython3-dev python3-numpy \
+    python3-decorator python3-opengl doxygen python3-sphinx python3-lxml ipython3
+\endverbatim
+
+If your system is Ubuntu (or compatible) and you are not interested in further
+details you can jump down to building at this point.
+
+The table below details all the build and runtime dependencies. For Python
+related packages, only the package names for version 3 are provided. For
+reference, the names of the rpm packages for Centos with the EPEL repository
+extensions are also provided (in this case all Python packages are 2.7).
+
+\htmlonly
+<table id="depends_table" style="width: 80%;">
```

Comment:
Is there a way to view a rendered version of this table?

Reply:
I'm afraid there isn't any automatic mode, unless you compile the documentation yourself